src/util/get-latest-package-version.ts

```diff
@@ -0,0 +1,26 @@
+import semver from "semver";
+
+export async function getLatestPackageVersion(packageName: string, majorVersion?: number): Promise<string | null> {
+    const url = `https://registry.npmjs.org/${packageName}`;
+
+    try {
+        const response = await fetch(url);
+        if (!response.ok) {
+            throw new Error(`Failed to fetch package info: ${response.status} ${response.statusText}`);
+        }
+
+        const packageInfo = (await response.json()) as { version: string; versions: { [version: string]: unknown } };
+
+        if (!majorVersion) {
+            return packageInfo.version;
+        }
+
+        const versions: string[] = Object.keys(packageInfo.versions);
+        const latestForMajor = versions.filter((version) => semver.satisfies(version, `^${majorVersion}.0.0`)).sort(semver.rcompare)[0];
+
+        return latestForMajor || null;
+    } catch (error) {
+        console.error(`Failed to fetch package info for ${packageName}: ${error}`);
+        return null;
+    }
```

Comment:
Why not throw the error here? If NPM is down, the install step won't work anyway?

Reply:
True f37b3454c53d8d080c22b0615895c5fba68f3cbd